python-backend/app/api/applications/resources/application.py

```diff
@@ -0,0 +1,154 @@
+
+        if not mine_guid:
+            raise BadRequest('A mine guid must be provided.')
+
+        applications = Application.find_by_mine_guid(mine_guid)
+
+        return applications
+
+    @api.expect(parser)
+    @api.doc(
+        description=
+        'This endpoint creates a new application for the mine specified using the form-data passed.',
+        responses={
+            400: 'Resource not found.',
+            404: 'Bad request.',
+        },
+        params={'mine_guid': 'A mine guid to associate the application to.'})
+    @api.marshal_with(application_model, code=201)
+    @requires_role_mine_create
+    def post(self):
+
+        data = self.parser.parse_args()
+
+        mine = Mine.find_by_mine_guid(data.get('mine_guid'))
+
+        if not mine:
+            raise NotFound('There was no mine found with the provided mine_guid.')
+
+        application_no, application_status_code, received_date = list(
+            map(data.get, ['application_no', 'application_status_code', 'received_date']))
+
+        if not application_no or not application_status_code or not received_date:
+            raise BadRequest(
+                'An application Number, Received Date and Status are required for an application.')
+
+        try:
+            application = Application.create(mine.mine_guid, data['application_no'],
+                                             data['application_status_code'], data['received_date'],
+                                             data.get('description'), self.get_create_update_dict())
+            application.save()
+        except Exception as e:
+            raise InternalServerError('Error: {}'.format(e))
+        return application, 201
+
+
+class ApplicationResource(Resource, UserMixin):
+    parser = reqparse.RequestParser()
+    parser.add_argument(
+        'application_status_code',
+        type=str,
+        help='Status of the application being added.',
+        store_missing=False)
+    parser.add_argument(
+        'description', type=str, help='Application description', store_missing=False)
+
+    @api.marshal_with(application_model, envelope='applications', code=200, as_list=True)
+    @api.doc(
+        description='This endpoint returns a single application based on its application guid.',
+        params={
+            'application_guid': 'Application guid to find a specific application.',
+        })
+    @requires_role_mine_view
+    def get(self, application_guid=None):
+        if not application_guid:
+            raise BadRequest('An application guid or a mine guid must be provided.')
+
+        application = Application.find_by_application_guid(application_guid)
+
+        if not application:
+            raise NotFound('Application not found')
+
+        return application
+
+    @api.expect(parser)
+    @api.doc(
+        description=
+        'This endpoint updates an application for the mine specified using the form-data passed.',
+        responses={
```

Comment:
This is purely for documentation correct, what are the expected error responses? Do we need to include the 200 response code here as well?

Reply:
The marshal_with handles the 200 or 201 or whatever the success response. The responses section is only for the error responses as far as I could tell.